lib/src/clients.dart

```diff
@@ -858,6 +859,7 @@ Future<http.StreamedResponse> _validateResponse(
   // TODO: We assume that status codes between [200..400[ are OK.
   // Can we assume this?
   if (statusCode < 200 || statusCode >= 400) {
+    @alwaysThrows
```

Comment:
Actually I think instead of doing this, it should return `Future.error()` instead.  It seems a little fishy that `_validateResponse()` can result in 3 different things (notice, it's not an `async {}` function):

  * a future completed with a value
  * a future completed with an error
  * a synchronous exception

 In practice the last one, a synchronous exception, doesn't happen, but it seems cleaner.